pkg/security/apparmor/validate.go

```diff
@@ -78,15 +78,31 @@ func (v *validator) Validate(pod *v1.Pod) error {
 		return fmt.Errorf("could not read loaded profiles: %v", err)
 	}
 
+	hasUnconfinedInitContainer := false
+
 	var retErr error
 	podutil.VisitContainers(&pod.Spec, podutil.AllContainers, func(container *v1.Container, containerType podutil.ContainerType) bool {
-		retErr = validateProfile(GetProfileName(pod, container.Name), loadedProfiles)
+		profileName := GetProfileName(pod, container.Name)
+
+		// It may be possible that the init container deploys the AppArmor
+		// profile which should be used by the workload. In this case we skip
+		// further validations and let the underlying container runtime decide.
+		if containerType == podutil.InitContainers && profileName == "" {
```

Comment:
I think I'm with @deads2k on this.

Why is the unconfined init container special? The annotation is per-container, just as seccomp is, albiet on the `container.securityContext`.

There is no seccomp admit handler in the kubelet IIRC.  Why shouldn't we just let all non-existent profile errors fall through to the runtime and result in `CreateContainerError` like seccomp?

Reply:
@sjenning I concur with the view that init containers should not be a special case. We are probably better off removing the checks against the _profile's existence_ altogether.

Reply:
I agree we better remove the check if the profile exists completely.